alerter/src/main.rs

```diff
@@ -0,0 +1,248 @@
+use anyhow::{Context, Result};
+use model::{
+    order::{Order, OrderCreation, OrderKind, OrderStatus, OrderUid},
+    u256_decimal,
+};
+use primitive_types::U256;
+use reqwest::Client;
+use std::time::{Duration, Instant, SystemTime};
+use url::Url;
+
+struct OrderBookApi {
+    base: Url,
+    client: Client,
+}
+
+impl OrderBookApi {
+    pub fn new(client: Client) -> Self {
+        Self {
+            base: "https://protocol-mainnet.gnosis.io".parse().unwrap(),
+            client,
+        }
+    }
+
+    pub async fn solvable_orders(&self) -> reqwest::Result<Vec<Order>> {
+        let mut url = self.base.clone();
+        url.set_path("/api/v1/solvable_orders");
+        self.client
+            .get(url)
+            .send()
+            .await?
+            .error_for_status()?
+            .json()
+            .await
+    }
+
+    pub async fn order(&self, uid: &OrderUid) -> reqwest::Result<Order> {
+        let mut url = self.base.clone();
+        url.set_path(&format!("/api/v1/orders/{}", uid));
+        self.client
+            .get(url)
+            .send()
+            .await?
+            .error_for_status()?
+            .json()
+            .await
+    }
+}
+
+struct MatchaApi {
+    base: Url,
+    client: Client,
+}
+
+impl MatchaApi {
+    pub fn new(client: Client) -> Self {
+        Self {
+            base: "	https://api.0x.org".parse().unwrap(),
```

Comment:
Is the additional white space here intended?
```suggestion
            base: "https://api.0x.org".parse().unwrap(),
```